R/backend-oracle.R

```diff
@@ -45,9 +45,26 @@ sql_translate_env.Oracle <- function(con) {
       as.integer64  = sql_cast("NUMBER(19)"),
       as.numeric    = sql_cast("NUMBER"),
       as.double     = sql_cast("NUMBER"),
+      #as.Date = sql_cast("DATE"),
+      # Oracle date casting is NLS dependant, the following will work for the default YYYY-MM-DD
+      as.Date = function(x) sql_expr("DATE " !!x),
```

Comment:
This doesn't look like valid syntax. Can you please double check?

Reply:
If you meant the R syntax you're right, not sure why I did it that way, it'll be corrected in my commit. 
The SQL syntax is the ANSI SQL string literal. At least in Oracle SQL (I can't speak for others) using `cast` is dependant on the NLS settings, which means unless you know this and format your date string in advance then you can't be certain it will work. the ANSI uses the YYYY-MM-DD format (which matches the default lubridate format string, at least in my localisation?). I thought this would be the safest way to catch most dates short of adding in and translating the format options?
The last script of the introduction section here shows this example https://oracle-base.com/articles/misc/oracle-dates-timestamps-and-intervals